test/test.jl

```diff
@@ -110,3 +109,110 @@ end
 
 # Test @test_approx_eq_eps
 # TODO
+
+ts = @testset "@testset should return the testset" begin
+    @test true
+end
+@test typeof(ts) == Base.Test.DefaultTestSet
+@test typeof(ts.results[1]) == Base.Test.Pass
+
+tss = @testloop "@testloop should return an array of testsets: $i" for i in 1:3
+    @test true
+end
+@test length(tss) == 3
+@test typeof(tss[1]) == Base.Test.DefaultTestSet
+@test typeof(tss[1].results[1]) == Base.Test.Pass
+
+# now we're done running tests with DefaultTestSet so we can go back to STDOUT
+redirect_stdout(OLD_STDOUT)
+
+# import the methods needed for defining our own testset type
+import Base.Test: record, finish, get_testset_depth, get_testset
```

Comment:
Ah, maybe to be more clear I should have `using Base.Test: get_testset_depth, get_testset, etc...` and only use `import` for the functions we're defining new methods on. This is less important here but good to set the right example in the docs I'm writing.